server/endpoints/workspaces.js

```diff
@@ -761,6 +763,91 @@ function workspaceEndpoints(app) {
       }
     }
   );
+
+  app.post(
+    "/workspace/:slug/thread/fork",
+    [validatedRequest, flexUserRoleValid([ROLES.all]), validWorkspaceSlug],
+    async (request, response) => {
+      try {
+        const user = await userFromSession(request, response);
+        const workspace = response.locals.workspace;
+        const { chatId, threadSlug } = reqBody(request);
+
+        if (!chatId) {
+          return response.status(400).json({ message: "chatId is required" });
+        }
+        const { thread: newThread, message: threadError } =
+          await WorkspaceThread.new(workspace, user?.id);
+
+        if (threadError) {
+          return response.status(500).json({ error: threadError });
+        }
+        const chatsToFork = await WorkspaceChats.where(
+          {
+            workspaceId: workspace.id,
+            user_id: user?.id,
+            thread_id: threadSlug
+              ? (
+                  await WorkspaceThread.get({
+                    slug: threadSlug,
```

Comment:
This is user input, cast it to the expected type to prevent injection via JSON schema `String(threadSlug)`